### PR TITLE
igr: environ2: Ignore `unset` errors for `UID`, `GID`

### PR DESCRIPTION
**Pull request by mobin-2008**
`Signed-off-by: Mobin "Hojjat" Aydinfar <mobin@mobintestserver.ir>`

### CODE REVIEW

File: src/igr-tests/environ2/run-test.sh [hunk 1]
```diff
@@ -14,11 +14,11 @@ RGID=$(id -g)
 for var in USER LOGNAME SHELL; do
     unset $var
 done
-# $UID & $GID are readonly on MacOS
-if [ "$(uname)" != "Darwin" ]; then
-    unset UID
-    unset GID
-fi
+# ignore $UID & $GID unsetting errors
+# Some systems make them readonly (such as MacOS, Void linux)
```
Comment (davmac314) [inline src/igr-tests/environ2/run-test.sh:18]: It's nothing to do with Void linux, it's caused by the shell that the tests are run with (eg. bash), please adjust the comment appropriately

**Review by davmac314 — CHANGES_REQUESTED** [review]

File: src/igr-tests/environ2/run-test.sh [hunk 2]
```diff
@@ -14,11 +14,11 @@ RGID=$(id -g)
 for var in USER LOGNAME SHELL; do
     unset $var
 done
-# $UID & $GID are readonly on MacOS
-if [ "$(uname)" != "Darwin" ]; then
-    unset UID
-    unset GID
-fi
+# ignore $UID & $GID unsetting errors
+# Some shells make them readonly
```
Comment (davmac314) [inline src/igr-tests/environ2/run-test.sh:18]: It would've been good to be specific here. Eg. "some shells - eg bash - make them readonly".

**Review by davmac314 — APPROVED** [review]

File: src/igr-tests/environ2/run-test.sh (hunk 2 above)
Comment (mobin-2008) [reply src/igr-tests/environ2/run-test.sh:18]: It actually based on system configuration for example I easily unset it on bash:
```bash
> echo $GID
1000
> unset GID
> echo $GID

>
```

Comment (mobin-2008) [reply src/igr-tests/environ2/run-test.sh:18]: (or maybe bash have different behavior in non-interactive mode?)

Comment (davmac314) [reply src/igr-tests/environ2/run-test.sh:18]: UID is readonly in bash. Bash doesn't set GID at all.
On MacOS, zsh sets UID and GID readonly.

**Pull request merged**